haskell.nix/application/weeder.toml

```diff
@@ -0,0 +1,2 @@
+roots = [ "^Main.main$", "^Paths_.*" ]
```

Comment:
@gromakovsky, does this look suitable for the default configuration?

Reply:
The default value for `roots` in `weeder`'s README is `[ "Main.main", "^Paths_weeder.*" ]`.
1. Why do we change specific `Main.main` to `^Main.main$`?
2. IIRC, `^Paths_.*` is used for auto-generated files and doesn't necessarily contain `Paths_weeder`, so this change looks reasonable.

Also why do we set `type-class-roots = true?

Reply:
>Why do we change specific Main.main to ^Main.main$?

Hmm, I think I just copied this from another [README section](https://github.com/ocharles/weeder?tab=readme-ov-file#calling-weeder).  The only difference is that `Main.main` can be a substring. Not sure if this will ever cause any problems. Would you like me to change it to `Main.main`?

> Also why do we set type-class-roots = true

IIRC I added it because I thought that sometimes a typeclass might have some functions that are already defined but not used yet, but now I think they should be handled manually. I'll remove this.

Reply:
Ok, I see. But I think strict `Main.main` is better. Imagine you have `mainWork` in `Main` and it's not (transitively) used by `main`. I think it should be considered dead code, shouldn't it?

Reply:
>I think it should be considered dead code, shouldn't it?

yes, I agree, but `^Main.main$` is more strict since it matches the beginning (`^`) and the end of the line (`$`), while `Main.main` can match a substring

Reply:
Oooh, I agree, I confused the syntax. Looks good then.

Reply:
> I'll remove this.

Seems to be the last thing to do.

Reply:
>Seems to be the last thing to do.

Done